core-bundle/src/OptIn/OptInInterface.php

```diff
@@ -0,0 +1,31 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of Contao.
+ *
+ * (c) Leo Feyer
+ *
+ * @license LGPL-3.0-or-later
+ */
+
+namespace Contao\CoreBundle\OptIn;
+
+interface OptInInterface
+{
+    /**
+     * Creates a new double opt-in token.
+     */
+    public function create(string $prefix, string $email, array $related): OptInTokenInterface;
+
+    /**
+     * Finds a double opt-in token by its identifier.
+     */
+    public function find(string $identifier): ?OptInTokenInterface;
+
+    /**
+     * Purges expired tokens.
+     */
+    public function purgeTokens(): void;
```

Comment:
Should name it `purgeExpiredTokens()` probably. `purgeTokens()` sounds like it would purge all of them.

Reply:
The word "purge" implies that you are only getting rid of what you no longer need and not that you throw away everything IMHO.